android/src/main/java/net/no_mad/tts/TextToSpeechModule.java

```diff
@@ -217,12 +304,32 @@ public void speak(String utterance, ReadableMap params, Promise promise) {
 
         String utteranceId = Integer.toString(utterance.hashCode());
 
-        int speakResult = speak(utterance, utteranceId, params);
-        if(speakResult == TextToSpeech.SUCCESS) {
-            promise.resolve(utteranceId);
-        } else {
-            resolvePromiseWithStatusCode(speakResult, promise);
-        }
+        // queue TTS utterance using single thread executor so that audio focus can be requested early
+        // when requesting to speak an utterance instead of waiting for it to be realy started to avoid
+        // nested requests when an utterance is requested to be spoken while another is still playing
+        executor.execute(() -> {
+            if (enableTestCode) {
+                testCodeReset();
+            }
+
+            isTtsCompleted = false;
```

Comment:
Is this thread safe?

Reply:
I think so yes. It is running in a single thread executor and all the code that sets it back to true is on response of what is running inside the executor.